htlcswitch/switch.go

```diff
@@ -445,55 +445,63 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, paymentID uint64,
 // NOTE: This function is synchronous and will block until either the
 // forwarding policies for all links have been updated, or the switch shuts
 // down.
-func (s *Switch) UpdateForwardingPolicies(newPolicy ForwardingPolicy,
-	targetChans ...wire.OutPoint) error {
-
-	log.Debugf("Updating link policies: %v", newLogClosure(func() string {
-		return spew.Sdump(newPolicy)
-	}))
+func (s *Switch) UpdateForwardingPolicies(
+	chanPolicies map[wire.OutPoint]*channeldb.ChannelEdgePolicy) error {
 
 	var linksToUpdate []ChannelLink
 
 	s.indexMtx.RLock()
 
 	// If no channels have been targeted, then we'll collect all inks to
 	// update their policies.
-	if len(targetChans) == 0 {
-		for _, link := range s.linkIndex {
-			linksToUpdate = append(linksToUpdate, link)
-		}
-	} else {
-		// Otherwise, we'll only attempt to update the forwarding
-		// policies for the set of targeted links.
-		for _, targetLink := range targetChans {
-			cid := lnwire.NewChanIDFromOutPoint(&targetLink)
-
-			// If we can't locate a link by its converted channel
-			// ID, then we'll return an error back to the caller.
-			link, ok := s.linkIndex[cid]
-			if !ok {
-				s.indexMtx.RUnlock()
-
-				return fmt.Errorf("unable to find "+
-					"ChannelPoint(%v) to update link "+
-					"policy", targetLink)
-			}
+	for targetLink := range chanPolicies {
+		cid := lnwire.NewChanIDFromOutPoint(&targetLink)
 
-			linksToUpdate = append(linksToUpdate, link)
+		// If we can't locate a link by its converted channel
+		// ID, then we'll log an error.
+		link, ok := s.linkIndex[cid]
+		if !ok {
+			log.Errorf("unable to find ChannelPoint(%v) to update "+
```

Comment:
Can just log as debug as it's not a fatal error since the link will have the right policy once it becomes active again. Log should also start capitalized.